pkgs/applications/editors/emacs/generic.nix

```diff
@@ -67,7 +67,29 @@ let emacs = (if withMacport then llvmPackages_6.stdenv else stdenv).mkDerivation
   pname = pname + lib.optionalString ( !withX && !withNS && !withMacport && !withGTK2 && !withGTK3 ) "-nox";
   inherit version;
 
-  patches = patches fetchpatch;
+  patches = patches fetchpatch ++
+            lib.optionals nativeComp
+              [
+                (substituteAll {
+                  src = let majorVersionStr = lib.versions.major version;
+                        in
+                          if lib.versionOlder version "29" || (lib.stringLength majorVersionStr == 8 && lib.versionOlder version < "20221003")
```

Comment:
I removed the `<`.

`srcRepo` is `true` by default, even without the overlay, and I don't see anyone overriding it to `false` in nixpkgs, so I don't think checking using `srcRepo` is what we want.

Reply:
I see, you are right! I remember it was set to `false` at some point, but it was changed a [few months ago](https://github.com/NixOS/nixpkgs/commit/54e795706d9fa90004ed1274af639a717f04a2aa). We should remove `srcRepo` since it's not effectively used.

IMHO we can leave out the `(lib.stringLength majorVersionStr == 8 && lib.versionOlder version "20221003")` part because it _only_ affects the `emacsGit` overlay and it will become dead code and difficult of maintain. But if you still want to keep it, please add a comment about it.

Besides that, LGTM.  Since this PR was closed, please open another PR and ping me and I'll merge it as soon as possible.